command/agent/cache/listener.go

```diff
@@ -48,21 +55,27 @@ func StartListener(lnConfig *configutil.Listener) (net.Listener, *tls.Config, er
 		}
 		ln, err = listenerutil.UnixSocketListener(addr, uConfig)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 
 	default:
-		return nil, nil, fmt.Errorf("invalid listener type: %q", lnConfig.Type)
+		return nil, fmt.Errorf("invalid listener type: %q", lnConfig.Type)
 	}
 
 	props := map[string]string{"addr": ln.Addr().String()}
-	tlsConf, _, err := listenerutil.TLSConfig(lnConfig, props, nil)
+	tlsConf, reloadFunc, err := listenerutil.TLSConfig(lnConfig, props, nil)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	if tlsConf != nil {
 		ln = tls.NewListener(ln, tlsConf)
 	}
 
-	return ln, tlsConf, nil
+	cfg := &ListenerBundle{
```

Comment:
I'm not convinced we need this struct, we could just return all three of the values of it, kind of like we were doing before, but with another return value, the reload func. I think it's a bit cleaner to read without, as `ListenerBundle` doesn't tell me a lot about what's inside. I think it'd be clearer without.

Reply:
This was based on the guidance from the internal engineering handbook regarding the maximum number of return values from a function before using a struct.

Reply:
At least in this case, I'd say that it's clearer without, but this is far from a hill I'm willing to die on.

Reply:
Just as a counter-point, I think the struct to bundle all these values together is valuable here, as it prevents sprawl of return values and simplifies what is actually being returned. There's less cognitive overhead with keeping track of X, Y, Z, etc. I've done similar things in the autopilot code.